src/controllers/components/cart-panel.js

```diff
@@ -0,0 +1,37 @@
+import { searchItem, checkIn} from '../../lib/api-client';
+import { Dispatcher } from 'consus-core/flux';
+import StudentStore from '../../store/student-store';
+
+export default class CartController {
+
+    static checkInItem(id, itemAddress) {
+        return checkIn(id, itemAddress).then(item => {
+            Dispatcher.handleAction('CHECKIN_SUCCESS', {
+                itemAddress: item.itemAddress
+            });
+        }).catch(error => {
+            Dispatcher.handleAction('ERROR', {
+                error
+            });
+        });
+    }
+
+    static getItem(address) {
+        if(StudentStore.getStudent().hasOverdueItem)
+            return Dispatcher.handleAction('ERROR', {
+                error:'Student has at least one overdue item.'
+            });
+
+        return searchItem(address).then(item => {
+            if (item.status === 'CHECKED_OUT')
+                return Dispatcher.handleAction('ERROR', {
+                    error: 'This item is already checked out by another student.'
+                });
+            Dispatcher.handleAction("CHECKOUT_ITEM_FOUND", item);
+        });
+    }
+
+    static throwError(error){
```

Comment:
Nit: I don't think this is necessary since calling `CartController.throwError('foo')` doesn't add any more context than simply doing `Dispatcher.handleAction('ERROR', { error: 'foo' });`.
On the other hand, change the name to something like "throwInvalidCharacterError".